cns/cnsclient/httpapi/client.go

```diff
@@ -11,7 +11,45 @@ type Client struct {
 }
 
 // UpdateCNSState updates cns state
-func (client *Client) UpdateCNSState(createNetworkContainerRequest *cns.CreateNetworkContainerRequest) error {
-	//Mat will pick up from here
-	return nil
+func (client *Client) UpdateCNSState(ipConfigs []*cns.ContainerIPConfigState) error {
+	var (
+		ipConfigsToAdd []*cns.ContainerIPConfigState
+	)
+
+	//Lock to read ipconfigs
+	client.RestService.Lock()
```

Comment:
It feels strange for this calling code to mess with the internals of its dependency (locking and mutating). Ideally RestService should not be exporting its mutex, but that's probably a larger change. Would these methods be better just as part of RestService? In that case, RestService still meets your defined interface, and that would be all you inject where necessary.

Reply:
+1, just pass array of ipconfigs to client

Reply:
@matmerr and I talked offline and matmerr is going to add a method to the RestService "GetIPConfigsFromState()" which returns inuseIPs. Request controller will then do the filtering by
`inuseIPs := client.RestService.GetIPConfigsFromState()`
client will then use this list of inuseIPs to sift out ipconfigs that cns already has, and thus, all locking/internal RestService things happen in RestService

Reply:
I think for now I am just going to merge in this version of UpdateCNSState and fix it later in a later PR after @matmerr has his changes in

Reply:
I agree with Ramiro, I dont think it is right to muck with internal state in CNS in the client code.